src/device_hub.cpp

```diff
@@ -52,14 +52,12 @@ namespace librealsense
         _ctx->set_devices_changed_callback({cb,  [](rs2_devices_changed_callback* p) { p->release(); }});
     }
 
-    std::shared_ptr<device_interface> device_hub::create_device(std::string serial)
+    std::shared_ptr<device_interface> device_hub::create_device(const std::string& serial, bool cycle_devices)
     {
-        for(auto i = 0; i< _device_list.size(); i++)
+        std::shared_ptr<device_interface> res = nullptr;
+        for(auto i = 0; ((i< _device_list.size()) && (nullptr == res)); i++)
```

Comment:
is this loop required at all if not requesting serial? if using _camera_index - no need to loop...

Reply:
The loop represents the general case. Requesting device without a serial will stop after one iteration provide the first* device available (*as designated by internal _camera_index)

Reply:
there is no real general case - serial requires loop. index requires direct access. 
serial has no loop_devices, index may have loop_devices.
I think unifying the 2 options in the first place hasn't much code share. but not critical.